target/linux/ath79/dts/qca95xx_dlink_dap-2xxx.dtsi

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "bdcfg";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				label = "rgdb";
+				reg = <0x50000 0x10000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "unknown";
+				reg = <0x60000 0x10000>;
+				read-only;
+			};
+
+			firmware: partition@70000 {
+				label = "firmware";
+				reg = <0x70000 0xee0000>;
+				compatible = "wrg";
+			};
+
+			dlink: partition@f50000 {
+				label = "dlink";
+				reg = <0xf50000 0xa0000>;
+				read-only;
+			};
```

Comment:
Is it acceptable to use aliases for certain partitions, to allow these to be overriden in the .dts when the flash layout for a device differs only in the ratio of these two partitions (i.e. DAP-2695) ?
Should we maybe prefix this with e.g. `mtd_` or `_partition` etc. ?

Reply:
Typically, I prefer to have the full partition table with the devices in the DTS files then, as everything else is typically much harder to read and thus increases the risk of having a wrong partition scheme (or updating it the wrong way later ...)

Reply:
You may keep part of the partition in DTSI, if only the "late" partitions differ, e.g.
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9563_tplink_tl-wpa8630p-v2.dtsi
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9563_tplink_tl-wpa8630p-v2-int.dts